p2p/src/types.rs

```diff
@@ -0,0 +1,473 @@
+//! Shared address types for p2p networking.
+
+use bytes::{Buf, BufMut};
+use commonware_codec::{EncodeSize, Error as CodecError, FixedSize, Read, ReadExt, Write};
+use commonware_runtime::{Error as RuntimeError, Resolver};
+use commonware_utils::{Hostname, IpAddrExt};
+use std::net::{IpAddr, SocketAddr};
+
+const INGRESS_SOCKET_PREFIX: u8 = 0;
+const INGRESS_DNS_PREFIX: u8 = 1;
+
+const ADDRESS_SYMMETRIC_PREFIX: u8 = 0;
+const ADDRESS_ASYMMETRIC_PREFIX: u8 = 1;
+
+/// What we dial to connect to a peer.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Ingress {
```

Comment:
The naming confused me because it seems "backwards"? Like isn't dialing considered a form of egress?